generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -137,15 +145,14 @@ public ClassTemplateSpec generate(DataSchema schema, DataSchemaLocation location
   }
 
   /**
-   * Generate {@link ClassTemplateSpec} from the specified {@link DataSchema} and its location.
-   *
-   * @param skipDeprecatedField flag to indicate whether to skip spec generation for deprecated fields, their types and referenced types.
-   *                            This behavior affect the spec generation recursively.
+   * Do not use this deprecated method. If need to skip deprecated field when generating specs, please use setSkipDeprecatedPatterns
+   * and generate(DataSchema schema, DataSchemaLocation location) instead.
    */
+  @Deprecated
   public ClassTemplateSpec generate(DataSchema schema, DataSchemaLocation location, boolean skipDeprecatedField)
   {
     pushCurrentLocation(location);
-    final ClassTemplateSpec result = processSchema(schema, null, null, skipDeprecatedField);
+    final ClassTemplateSpec result = processSchema(schema, null, null);
```

Comment:
This deprecated method should call the non-deprecated method, not vice versa (core logic should be moved out of deprecated methods marked for deletion).